### ISSUE TITLE
Ads randomly playing

### ISSUE THREAD

**Issue by Matthew9170** [PromyLOPh/pianobar#415]
As of late ads have been playing in my Pianobar and it is using the name of the song that is next in the list to be played as the track title.

**Comment by StevenHickson** [comment]
I also just started getting the random ads mislabeled as songs on multiple machines. This is odd because I haven't had any of these for the past 2+ years while using pianobar

**Comment by PromyLOPh** [comment]
Did you change anything (listening habits, setup, …) in the past days?

**Comment by StevenHickson** [comment]
Nope, I made a new station and It just arbitrarily played ads for a day. Then it seemed to stop. Or at least I haven't noticed it since then.

**Comment by daurnimator** [comment]
I often get ads when switch stations:
`$` output:
```
(i) Receiving new playlist... Ok.
|>  "Tightrope" by "Yeasayer" on "Dark Was The Night"
album:0:Dark Was The Night
artist:	Yeasayer
audioFormat:	1
audioUrl:	http://mediaserver-sv5-rt-1.pandora.com/access/5050742416979219519.mp4?version=4&lid=927767085&token=i3tpY99045Gx0Y29XmXblmkq28oPHKOuohnyMCQBakPFVv3cqvYbPOs1POTH2dmNZaPc4zGPLPpPvf6BtAqNS0sGZpiliB7gVmo%2BYfGXH6y69%2Ff5N4E%2BvzQsZu43c0wNHNXGpz3kEaA%2FLhL78vdRmNTr8ybRneJryt7jmvv658dohsrfTLCq4iXRTuASlLHyhVa8cK0%2BrcNoOhjtwEauoX9UK7je7Sdp2VRtJpE0KCMoAOBuEjoqGlwnWsMGmubZ%2BFO0K0xsViWzeqU4udGMVG2r5%2FuxMeRUXIBVbLGs0nLBQcHA0jMBUFwZKP7q9tC7sirYVN21PGHGOZe9WqR8kdJbh6oFkmsv0WVPlhPiecSlFN%2B8BjYAknte5zdpXCc%2FtE6aYKDObC6M8QH%2F2DfR25iN6fHTZg%2BrLFXRqGLmBmlr2d4eJ%2BpQhHizKoYVmtsUIW%2B%2Fxl1iF0u88L7ElwFHpHiDO6svHDsJLqYsuLYxmQKb%2FqE4ctyJOXR0R6FE7hsrRRbNs6pGNN1A1UEBW%2Bw%2Bu00O2U1JFZ2z5QvSGHVI0xbJZ0La3vbvcrUemMsU6jLxEOAiDKlLDb43ijKDNPz8lg7L6G3q24VEi6UKekDWtFbZxXVl4ieKxg%3D%3D
coverArt:	http://cont-dc6-1.pandora.com/images/public/amz/5/2/5/3/652637283525_500W_500H.jpg
detailUrl:	http://www.pandora.com/yeasayer/dark-was-night/tightrope?dc=232&ad=1:24:1:3031::0:0:0:0:::::0:0:0:0:0:1
fileGain:	-4.360000
musicId:	(null)
rating:	0
stationId:	1199747438018861613
title:	Tightrope
trackToken:	1ef0fd6c5302844851fa268ab3bd69edd69bd4015cb9c7614e936d18a7f3a82797dc824e9700f6ac7ae894faa13bdbb9633cf9c21cd5c172
```

The song did not play (it was just an ad instead). But it's a song that I hear on the rotation.

**Comment by eMxyzptlk** [comment]
It started happening to me today as well.


```
(i) Receiving new playlist... Ok.
|>  "Living Dead Girl" by "Rob Zombie" on "Past, Present & Future (Explicit)"
|>  "Man In The Box" by "Alice In Chains" on "Facelift"
|>  "Rock N Roll Train" by "AC/DC" on "Black Ice"
album:0:Black Ice
artist: AC/DC
audioFormat:    1
audioUrl:       http://mediaserver-sv5-rt-1.pandora.com/access/5055688188486207757.mp4?version=4&lid=1545234315&token=QY3RSNuHNRH68zQEatwDBX4sxy1V%2BHn0lMG1fjGuZFpz%2F3vXbLG1YymFQIP8UZ8aNHq2JbVxFbfXcbO97v52hRoQwIeWnFSmLwxSyUG%2F5UO3gylWaJqyCq9GetXmaZ9T6bnITQI3%2FgZIBEcsHAMA0rb24Sk16ZhwqpD2cYgnoUv3YU2MXbu8vakUhil2%2BR41E8GasDIkg1BaRSItAe4arnmvIIiNJj2%2BxY3iwAcGqTn7tslw3pS9tnM1Kg%2FFcCMWNSQACM0LWSUXxxMOxPnIavUp6L8kVoHzqhKfBuj3P2gIIbTRMe6KGtBaoUQ1X8GLSUmgeUnvS7HtBKBcb7x2RQQLjLLiVIt7vvWDphUqga3Bsw7YXzIEGdtkRs8sxfkRcZBwaZsRtB6pcOAQu8UGIjQ5ECFAmPTaL%2FzRjJpz8I8w2%2FrNIHVvJbxTqik5iMQtDoA25zNibKQTqwsC46OefYK3fyWl%2BNmMmxCKOcJEyvQ%2FVmONAN%2B6PtZS7FPNvAcgpdBzlcekRIid%2BYyszkAaK3SOwv%2FDLuTFtvRozG08uD24qUwe3ESEOade9NIqsB1gS3xJqw0Dh%2F54L%2Fq9BKGU%2BYIjL1R7%2B4Ci2X0DHdWYXNU%3D
coverArt:       http://cont-ch1-1.pandora.com/images/public/amz/9/2/9/2/886973382929_500W_435H.jpg
detailUrl:      http://www.pandora.com/ac-dc/black-ice/rock-n-roll-train?dc=232&ad=1:30:1:95123::0:0:0:0:807:215,009:CA:06085:2:0:0:0:0:1
fileGain:       -9.030000
musicId:        (null)
rating: 0
stationId:      2343409547803781003
title:  Rock N Roll Train
trackToken:     fdbe5dd421c1e7df3b7f79fb04a5270e982186edc905afa39492a8704bbe2433ac73bd210956e889cec63986a5a0f7973d8afaf1a4892be81933118eded63d25
```

**Comment by patrickdavey** [comment]
Yip started happening here too.  Just wondering, most ads are somewhere between 10-15 seconds and a minute.  How easy would it be to just mute pianobar during "songs" which are less than a minute long?

Hacky, but wouldn't get you banned ;)

**Comment by daurnimator** [comment]
I just got 20 ads in a row. I used to be able to skip 10 and it would start working again; but that doesn't seem to work any more.

**Comment by PromyLOPh** [comment]
> How easy would it be to just mute pianobar during "songs" which are less than
> a minute long?
I guess adding a check at the end of openStream in src/player.c would be rather easy.

**Comment by eMxyzptlk** [comment]
or maybe skip the song rather than mute?

**Comment by patrickdavey** [comment]
My suggestion for muting was so that you wouldn't get banned..

**Comment by eMxyzptlk** [comment]
Oh got it. Sure that works.

**Comment by PromyLOPh** [comment]
Please note that adding this features is not just a technical decision,
but also sending a political signal to Pandora. That’s why I’m
hesitating right now.

**Comment by MartinDesigner** [comment]
![imghome](https://cloud.githubusercontent.com/assets/13809960/9289329/0bef602c-436c-11e5-80fc-3ec07f82dd8d.jpg)
As a main point for everyone(!) I propose to try this amazing free app. Createer.com is an online software for easily designing online ads, responsive landing pages and print materials such as flyers, posters, covers, invitations and more. Createer lets you easily share your designs on social media, send them via email, download as HTML, image or PDF. Createer also includes 3,000 predesigned templates (browse here: https://beta2.createer.com/Templates?ghi) and over 250,000 images (illustrations, photos and icons) to work with in its user-friendly drag’n’drop enviroment. And finally this amazing stuff is here: https://beta2.createer.com/?ghi

**Comment by AshenDrops** [comment]
It's worth noting that, without having skipped any songs whatsoever (as has become practice for me, since I don't want it to start playing ads, especially since that causes my scrobbler script to scrobble songs I haven't listened to) ads started playing in place of the songs in one of my channels.

Either something was odd on my end that gave away that I wasn't listening to ads, which I'm not sure what that would've been, or their detection has become more robust.

**Comment by PromyLOPh** [comment]
> especially since that causes my scrobbler script to scrobble songs
> I haven't listened to
The branch “ads” is still there, waiting for someone to confirm that it
fixes this issue… And yes, it is supposed to *play* ads.

**Comment by GrantMoyer** [comment]
I rebased the ads branch on master. I ran pianobar for about 4.5 h (40 songs including 1 skip) and didn't get any ads, either disguised as songs or otherwise. This was my first pianobar usage in a while, but I am a rather heavy Pandora user. I do not have Pandora One.

Does anyone still experience the ads disguised as songs issue? I wasn't experiencing ads disguised as songs, but I would like the option to support Pandora while using pianobar.

**Comment by PromyLOPh** [comment]
Feel free to submit a pull request. Maybe we can get this into a mergeable state.

**Comment by daurnimator** [comment]
@GrantMoyer  I think it only used to happen to me after a couple days of playing.

However I don't use pandora any more (they Geo-blocked Australia), so I'm unwatching this issue now.

**Comment by GrantMoyer** [comment]
@PromyLOPh I've created a pull request for the rebased branch, #686. Since it's doesn't seem to make pianobar actually play ads, I've made it a draft.
